exasol/error/_parse.py

```diff
@@ -218,40 +293,38 @@ def normalize(params):
                     yield k.value, ""
 
         return ErrorCodeDetails(
-            identifier=code.value,
-            message=message.value,
+            identifier=validated_node.code.value,
+            message=validated_node.message.value,
             messagePlaceholders=[
                 Placeholder(name, description)
-                for name, description in normalize(parameters)
+                for name, description in normalize(
+                    validated_node.parameters
+                )
             ],
             description=None,
             internalDescription=None,
             potentialCauses=None,
-            mitigations=(
-                (
-                    [m.value for m in mitigations.elts]
-                    if isinstance(mitigations, ast.List)
-                    else [mitigations.value]
-                )
-                if not isinstance(mitigations, str)
-                else [mitigations]
-            ),
+            mitigations=self._construct_mitigations(validated_node.code, validated_node.mitigations),
             sourceFile=self._filename,
-            sourceLine=node.lineno,
+            sourceLine=validated_node.lineno,
             contextHash=None,
         )
 
     def collect(self) -> None:
         for node in _ExaErrorNodeWalker(self._root):
-            errors, warnings = self._validator.validate(node, self._filename)
-            if errors:
+            validatation_result = self._validator.validate(
+                node, self._filename
+            )
+            if validatation_result.errors:
                 # stop if we encountered any error
                 return
-            error_definition = self._make_error(node)
+            if  validatation_result.node is None:
+                raise Exception("Validation finished with invalid result, but no error was set")
```

Comment:
I think, for that, we need one of these StaticErrors because they are the Error Code used in this project. Basically, the project can't analyze it self.